guides/provider/README.md

```diff
@@ -1,330 +1,458 @@
 
-### Tools
+## Kubernetes Cluster Setup
 
-Installed commands:
+The setup of a Kubernetes cluster is the responsibility of whoever sets up a provider on the Akash network. This section of this document provides best practices and  ecommendations for setting up a Kubernetes cluster.  This document is not a comprehensive guide to operating a Kubernetes cluster.
 
-* [`akash`](https://github.com/ovrclk/akash/releases)
-* [`kubectl`](https://kubernetes.io/docs/tasks/tools/install-kubectl)
-  * [`kustomize`](https://kubernetes-sigs.github.io/kustomize/installation/)
-* [`jq`](https://stedolan.github.io/jq/)
-* [`curl`](https://curl.haxx.se/)
+The general set of actions which must be carried out to setup a Kubernetes cluster for
+a provider is 
 
-### Working Directory
+1. Clone the Kubespray project
+2. Setup your Ansible inventory for the nodes that makeup the cluster
+3. Run the provisioning using ansible
+4. Retrieve the Kubernetes configuration file so that the cluster can be accessed
+5. Add Akash's Custom Resource Definitions to Kubernetes
+6. Add the NGINX Ingress controller
 
-Create staging directory and cd into it
+At ths point you would be left with a Kubernetes cluster that is ready to be a provider but not yet on the network.
 
-```sh
-mkdir akash-demo && cd akash-demo
-```
+The recommended method for setting up a Kubernetes cluster is to use the [Kubespray](https://github.com/kubernetes-sigs/kubespray) project. This project is a collection of ansible resources for setting up a Kubernetes cluster. 
 
-### Kubernetes Cluster
+The recommended minimum number of machines is three. One machine hosts the Kubernetes master node & provider, with the other machines hosting the compute nodes. It is possible however to provision a single-machine cluster if you
+choose to, but this configuration is not recommended.
 
-* Version: `1.19+`
-* A Kubernetes cluster with your local `kubectl` with admin authentication and authorization.
-* A wildcard A Record resolving to a cluster route-able IP. eg: `*.akashian.io`
+## Getting kubespray & setup
 
-#### Option 1: [Akash Kubespray](https://github.com/ovrclk/kubespray/) 
+Get the [latest](https://github.com/kubernetes-sigs/kubespray/releases) official release of Kubespray.
 
-Akash has created an example Kubespray configuration to provision a Kubernetes installation on linux machines. Included are best practice configuration for CNI `calico` plugin, default Network Policies + Namespaces, `nginx` Ingress Controller, `metrics-server`, `cert-manager`, and Seccomp syscall restrictions.
+You need a working Ansible installation as well, with the additional Python dependencies specified in the `requirements.txt` document. It is recommended you install this using a Python virtual environment created with the `virtualenv` tool.
 
+```
+virtualenv --python=python3 venv
+```
+```
+. venv/bin/activate
+```
+```
+sudo pip3 install -r requirements.txt
```

Comment:
After doing `activate` you should not need to use `sudo` in any context for running python. The `pip` command installs things into the virtual environment, not the machines root filesystem